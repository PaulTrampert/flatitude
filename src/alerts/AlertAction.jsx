import React from 'react';
import PropTypes from 'prop-types';

class AlertAction extends React.Component {

  handleActionClick = () => {
    let {
      alertId,
      onClick
    } = this.props;

    onClick({alertId});
  }

  render() {
    let {
      label
    } = this.props;

    return (
      <a className="action" onClick={this.handleActionClick}>
        {label}
      </a>
    );
  }
}

AlertAction.propTypes = {
  alertId: PropTypes.number,
  label: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default AlertAction;