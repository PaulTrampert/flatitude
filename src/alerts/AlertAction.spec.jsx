import React from 'react';
import {shallow} from 'enzyme';
import AlertAction from './AlertAction.jsx';

describe('AlertAction', () => {
  let subject;
  let onClick;

  beforeEach(() => {
    onClick = jest.fn();
    subject = shallow(<AlertAction label="herp" onClick={onClick} alertId={4} />);
  });

  describe('handleActionClick', () => {
    it('calls props.onClick with the action id', () => {
      subject.instance().handleActionClick();

      expect(onClick).toHaveBeenCalledWith({alertId: 4});
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('render', () => {

    it('renders the link', () => {
      expect(subject).toMatchSnapshot();
    });

    it('binds handleActionClick to the onClick event of the a tag', () => {
      expect(subject.find('a').props().onClick).toBe(subject.instance().handleActionClick);
    });
  });
});