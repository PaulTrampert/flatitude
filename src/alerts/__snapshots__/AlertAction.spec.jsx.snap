// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertAction render renders the link 1`] = `
<a
  className="action"
  onClick={[Function]}
>
  herp
</a>
`;
